### ISSUE TITLE
[BUG]: why use action_mask = action_mask[:, 1:]

### ISSUE THREAD

**Issue by shyoulala** [hpcaitech/ColossalAI#3217]
### 🐛 Describe the bug

在chatgpt的actor.py中，我看到返回action_mask的时候  使用了 action_mask = action_mask[:, 1:]，为什么要排出第0个，我认为不能排出第0个，我们返回的应该就生成的句子的mask呀？这里排出第0个不就少了一位吗？是我理解有问题吗？谢谢。

### Environment

The generate method for actor.py in chatgpt code，I see action mask = action mask[:, 1:]，

**Comment by Camille7777** [comment]
hii, @shyoulala , `action_mask = action_mask[:, 1:]`  since the relative position of sequence and logits is shifted by 1, we need remove the first column to make sure the mask matching corresponding logits.
